### ISSUE TITLE
Graphiql Docs with HapiJS

### ISSUE THREAD

**Issue by RobAWilkinson** [apollostack/graphql-server#191]
Queries are running perfectly fine with graphql and graphiql with hapiJS but the docs do not load in graphiqlHapi.

plugins are set up as follows:

```JS
  server.register({
      register: apolloHapi,
      options: {
        path: '/graphql',
        apolloOptions: {
          schema: executableSchema,
        },
      },
  });
  server.register({
      register: graphiqlHapi,
      options: {
        path: '/graphiql',
        graphiqlOptions: {
          endpointURL: '/graphql',
        },
        route: {
          cors: true
        }
      },
  });
```

The inital load on the `graphiql` route fires a post request to `/graphql?` with a really long query, my graphql responds with a `400` status code and a json response

```JSON
{  
   "errors":[  
      {  
         "message":"Fragment \"FullType\" cannot be spread here as objects of type \"__Type\" can never be of type \"__Type\".",
         "locations":[  
            {  
               "line":7,
               "column":9
            }
         ]
      },
      {  
         "message":"Fragment \"InputValue\" cannot be spread here as objects of type \"__InputValue\" can never be of type \"__InputValue\".",
         "locations":[  
            {  
               "line":13,
               "column":11
            }
         ]
      }
   ]
}
```

**Comment by stubailo** [comment]
What version of the "graphql" package are you using?

**Comment by RobAWilkinson** [comment]
0.7.2

**Comment by RobAWilkinson** [comment]
the introspection query graphiql fires is
 
```JS
query IntrospectionQuery {
    __schema {
      queryType { name }
      mutationType { name }
      subscriptionType { name }
      types {
        ...FullType
      }
      directives {
        name
        description
        locations
        args {
          ...InputValue
        }
      }
    }
  }

  fragment FullType on __Type {
    kind
    name
    description
    fields(includeDeprecated: true) {
      name
      description
      args {
        ...InputValue
      }
      type {
        ...TypeRef
      }
      isDeprecated
      deprecationReason
    }
    inputFields {
      ...InputValue
    }
    interfaces {
      ...TypeRef
    }
    enumValues(includeDeprecated: true) {
      name
      description
      isDeprecated
      deprecationReason
    }
    possibleTypes {
      ...TypeRef
    }
  }

  fragment InputValue on __InputValue {
    name
    description
    type { ...TypeRef }
    defaultValue
  }

  fragment TypeRef on __Type {
    kind
    name
    ofType {
      kind
      name
      ofType {
        kind
        name
        ofType {
          kind
          name
          ofType {
            kind
            name
            ofType {
              kind
              name
              ofType {
                kind
                name
                ofType {
                  kind
                  name
                }
              }
            }
          }
        }
      }
    }
  }
```

I'm seeing users of other packages having problems with this as well, there is some mention of snake_case vs camelCase in schema properties. too

**Comment by stubailo** [comment]
It might be some mismatch of the `graphql` package and `graphiql` versions - maybe the introspection query being sent is incompatible...

**Comment by RobAWilkinson** [comment]
after digging through it and going back to a brand new fresh state with a simple `HelloWorld` schema I still receive the same error. Seems like the particular schema doesnt matter, the camelCase was a wild goose chase

**Comment by RobAWilkinson** [comment]
great catch @stubailo a simple `yarn upgrade apollo-server graphql` fixed this.

**Comment by stubailo** [comment]
@RobAWilkinson how did going back to a fresh state not solve it? Is there something we should do in the packages to make this less likely?

**Comment by RobAWilkinson** [comment]
Sorry I should explain:

I rolled back to an empty schema in the hopes that there was a misplaced property or i was camel caseing when i should have been snake caseing.

I did not start a brand new project and do a fresh install.

@stubailo At some point my graphql, graphiql and apollo-server packages got out of sync. Someone who understands npm better than me can probably answer what the solution is. I'm not sure if that is to set a hard peer dependency or what.
